### PR TITLE
Add unit tests for ticket #58392

### PR DESCRIPTION
**Pull request by Hug0-Drelon**
Trac ticket: https://core.trac.wordpress.org/ticket/58392

This PR adds unit tests to the previously proposed diff.

### CODE REVIEW

File: tests/phpunit/tests/post/wpTrashPost.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
+<?php
+
```
Comment (audrasjb) [inline tests/phpunit/tests/post/wpTrashPost.php:2]: I think we need a docblock here:
```
/**
 * @group post
 * @covers ::wp_trash_post
 */
```
See https://github.com/WordPress/wordpress-develop/blob/trunk/tests/phpunit/tests/post/wpPublishPost.php

File: tests/phpunit/tests/post/wpTrashPost.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	public function set_up() {
+		parent::set_up();
+
+		$this->post = $this->factory()->post->create_and_get(
+			array(
+				'post_status' => 'draft',
+			)
+		);
+	}
+
```
Comment (audrasjb) [inline tests/phpunit/tests/post/wpTrashPost.php:23]: And a mention of the related ticket for each test case. For example:
```
	/**
	 * @ticket 58392
	 * @covers ::wp_trash_post
	 */
```

See https://github.com/WordPress/wordpress-develop/blob/trunk/tests/phpunit/tests/post/wpPublishPost.php

**Review by audrasjb — CHANGES_REQUESTED** [review]
Everything looks good to me, except that we need a few docblocks in the provided unit tests. 
You can take [wpPublishPost](https://github.com/WordPress/wordpress-develop/blob/trunk/tests/phpunit/tests/post/wpPublishPost.php) as an example.

File: tests/phpunit/tests/post/wpTrashPost.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	public function test_trash_post() {
```
Comment (costdev) [inline tests/phpunit/tests/post/wpTrashPost.php:32]: Thanks for these tests @Hug0-Drelon!

These should ideally be separate test methods to test each filter/action independently. That way, if there's a failure with one of them, the others will still be tested. Same for the `wp_untrash_post()` tests.

**Comment by Hug0-Drelon** [comment]
Hi @audrasjb and @costdev 👋 
Thanks for the feedback!
I've split the test and added some docblocks as you suggested.

File: tests/phpunit/tests/post/wpTrashPost.php [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+		$this->post = $this->factory()->post->create_and_get(
+			array(
+				'post_status' => 'draft',
+			)
+		);
+	}
+
+	/**
+	 * @ticket 58392
+	 * @covers ::wp_trash_post
```
Comment (costdev) [inline tests/phpunit/tests/post/wpTrashPost.php]:
```suggestion
	 * Tests that wp_trash_post() returns a WP_Post object
	 * and sets the correct post meta to trash a post.
	 *
	 * @ticket 58392
	 *
	 * @covers ::wp_trash_post
```
- Let's add descriptions to each of the tests, starting with "Tests that" per convention.
- I also spaced out the `@ticket` and `@covers` annotations based on [the structure/grouping of annotations in this ticket](https://core.trac.wordpress.org/ticket/56070).

File: tests/phpunit/tests/post/wpTrashPost.php [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+		$this->assertArrayHasKey( '_wp_trash_meta_status', $trashed_post_metas, 'Trashed post should have _wp_trash_meta_status meta set.' );
+		$this->assertCount( 1, $trashed_post_metas['_wp_trash_meta_status'], 'Trashed post should have only one _wp_trash_meta_status meta set.' );
+		$this->assertSame( $this->post->post_status, reset( $trashed_post_metas['_wp_trash_meta_status'] ), 'Trashed post should have _wp_trash_meta_status meta set to previous post status.' );
+		$this->assertArrayHasKey( '_wp_trash_meta_time', $trashed_post_metas, 'Trashed post should have _wp_trash_meta_time meta set.' );
+		$this->assertCount( 1, $trashed_post_metas['_wp_trash_meta_time'], 'Trashed post should have only one _wp_trash_meta_time meta set.' );
+	}
+
+	/**
+	 * @ticket 58392
+	 * @covers ::wp_trash_post
```
Comment (costdev) [inline tests/phpunit/tests/post/wpTrashPost.php]:
```suggestion
	 * Tests that wp_trash_post() applies 'pre_trash_post' filters
	 * and passes the expected values to callbacks.
	 *
	 * @ticket 58392
	 *
	 * @covers ::wp_trash_post
```

File: tests/phpunit/tests/post/wpTrashPost.php [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+		);
+
+		wp_trash_post( $this->post->ID );
+
+		$this->assertGreaterThan( 0, did_filter( 'pre_trash_post' ), 'pre_trash_post filter was not called.' );
+	}
+
+	/**
+	 * @ticket 58392
+	 * @covers ::wp_trash_post
```
Comment (costdev) [inline tests/phpunit/tests/post/wpTrashPost.php]:
```suggestion
	 * Tests that wp_trash_post() triggers the 'wp_trash_post' action
	 * and passes the expected values to callbacks.
	 *
	 * @ticket 58392
	 *
	 * @covers ::wp_trash_post
```

File: tests/phpunit/tests/post/wpTrashPost.php [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+		);
+
+		wp_trash_post( $this->post->ID );
+
+		$this->assertGreaterThan( 0, did_action( 'wp_trash_post' ), 'wp_trash_post action was not called.' );
+	}
+
+	/**
+	 * @ticket 58392
+	 * @covers ::wp_trash_post
```
Comment (costdev) [inline tests/phpunit/tests/post/wpTrashPost.php]:
```suggestion
	 * Tests that wp_trash_post() triggers the 'trashed_post' action
	 * and passes the expected values to callbacks.
	 *
	 * @ticket 58392
	 *
	 * @covers ::wp_trash_post
```

File: tests/phpunit/tests/post/wpUntrashPost.php [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+				array(
+					'post_status' => 'draft',
+				)
+			)
+		);
+	}
+
+	/**
+	 * @ticket 58392
+	 * @covers ::wp_untrash_post
```
Comment (costdev) [inline tests/phpunit/tests/post/wpUntrashPost.php]:
```suggestion
	 * Tests that wp_untrash_post() returns a WP_Post object,
	 * removes post meta for an untrashed post and sets it to a 'Draft'.
	 *
	 * @ticket 58392
	 *
	 * @covers ::wp_untrash_post
```

File: tests/phpunit/tests/post/wpUntrashPost.php [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+		$this->assertArrayNotHasKey( '_wp_trash_meta_time', $untrashed_post_metas, 'Untrashed post should not have _wp_trash_meta_time meta anymore.' );
+
+		$post = get_post( $this->trashed_post->ID );
+
+		$this->assertSame( 'draft', $post->post_status, 'Untrashed post should have its previous status set correctly.' );
+	}
+
+	/**
+	 * @ticket 58392
+	 * @covers ::wp_untrash_post
```
Comment (costdev) [inline tests/phpunit/tests/post/wpUntrashPost.php]:
```suggestion
	 * Tests that wp_untrash_post() applies 'pre_untrash_post' filters
	 * and passes the expected values to callbacks.
	 *
	 * @ticket 58392
	 *
	 * @covers ::wp_untrash_post
```

File: tests/phpunit/tests/post/wpUntrashPost.php [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+		);
+
+		wp_untrash_post( $this->trashed_post->ID );
+
+		$this->assertGreaterThan( 0, did_filter( 'pre_untrash_post' ), 'pre_untrash_post filter was not called.' );
+	}
+
+	/**
+	 * @ticket 58392
+	 * @covers ::wp_untrash_post
```
Comment (costdev) [inline tests/phpunit/tests/post/wpUntrashPost.php]:
```suggestion
	 * Tests that wp_untrash_post() triggers the 'untrash_post' action
	 * and passes the expected values to callbacks.
	 *
	 * @ticket 58392
	 *
	 * @covers ::wp_untrash_post
```

File: tests/phpunit/tests/post/wpUntrashPost.php [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+		);
+
+		wp_untrash_post( $this->trashed_post->ID );
+
+		$this->assertGreaterThan( 0, did_action( 'untrash_post' ), 'untrash_post action was not called.' );
+	}
+
+	/**
+	 * @ticket 58392
+	 * @covers ::wp_untrash_post
```
Comment (costdev) [inline tests/phpunit/tests/post/wpUntrashPost.php]:
```suggestion
	 * Tests that wp_untrash_post() triggers the 'untrashed_post' action
	 * and passes the expected values to callbacks.
	 *
	 * @ticket 58392
	 *
	 * @covers ::wp_untrash_post
```

File: tests/phpunit/tests/post/wpTrashPost.php [hunk 12]
```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * @group post
+ * @covers ::wp_trash_post
```
Comment (costdev) [inline tests/phpunit/tests/post/wpTrashPost.php]:
```suggestion
 * @group post
 *
 * @covers ::wp_trash_post
```

File: tests/phpunit/tests/post/wpTrashPost.php [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+ */
+class Tests_Post_wpTrashPost extends WP_UnitTestCase {
```
Comment (costdev) [inline tests/phpunit/tests/post/wpTrashPost.php]:
```suggestion
class Tests_Post_WpTrashPost extends WP_UnitTestCase {
```
`wp` should be `Wp` here. [See the handbook entry on Test Classes](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#test-classes) and [this ticket about 'wp' vs 'Wp' for test class names](https://core.trac.wordpress.org/ticket/56846).

File: tests/phpunit/tests/post/wpUntrashPost.php [hunk 14]
```diff
@@ -0,0 +1,115 @@
+<?php
+
+/**
+ * @group post
+ * @covers ::wp_untrash_post
```
Comment (costdev) [inline tests/phpunit/tests/post/wpUntrashPost.php]:
```suggestion
 * @group post
 *
 * @covers ::wp_untrash_post
```

File: tests/phpunit/tests/post/wpUntrashPost.php [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+ */
+class Tests_Post_wpUntrashPost extends WP_UnitTestCase {
```
Comment (costdev) [inline tests/phpunit/tests/post/wpUntrashPost.php]:
```suggestion
class Tests_Post_WpUntrashPost extends WP_UnitTestCase {
```

**Review by costdev — CHANGES_REQUESTED** [review]
Thanks for the updates @Hug0-Drelon! I've left some more thoughts in this review 🙂

**Comment by Hug0-Drelon** [comment]
Thanks for the insight @costdev! Nice to see how standardization is going 🙂

**Review by costdev — APPROVED** [review]
Thanks for the updates @Hug0-Drelon! I think that's all from me on this one!

**Review by audrasjb — APPROVED** [review]
Thanks for the update!

**Comment by audrasjb** [comment]
committed in https://core.trac.wordpress.org/changeset/56043

**Pull request closed without merging**